### ISSUE TITLE
Any value in mx.model.FeedForward.create ->  array.batch.size yields to "Check failed: from.shape() == to->shape() operands shape mismatchfrom.shape = (20,1) to.shape=(20,)"

### ISSUE THREAD

**Issue by chavarinbonbyn** [apache/incubator-mxnet#5538]
## Environment info
Operating System:
Linux Fedora 23 -  cat /proc/version 
Linux version 4.8.13-100.fc23.x86_64 (mockbuild@bkernel02.phx2.fedoraproject.org) (gcc version 5.3.1 20160406 (Red Hat 5.3.1-6) (GCC) ) #1 SMP Fri Dec 9 14:51:40 UTC 2016

Compiler:
gcc version 5.3.1 20160406

Package used (Python/R/Scala/Julia):
R version 3.3.1

MXNet version:
packageVersion("mxnet")
[1] ‘0.9.4’

If you are using R package, please provide
R `sessionInfo()`:
> sessionInfo()
R version 3.3.1 (2016-06-21)
Platform: x86_64-redhat-linux-gnu (64-bit)
Running under: Fedora 23 (Workstation Edition)

locale:
 [1] LC_CTYPE=es_MX.UTF-8       LC_NUMERIC=C               LC_TIME=es_MX.UTF-8        LC_COLLATE=es_MX.UTF-8    
 [5] LC_MONETARY=es_MX.UTF-8    LC_MESSAGES=es_MX.UTF-8    LC_PAPER=es_MX.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C             LC_MEASUREMENT=es_MX.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] mxnet_0.9.4

loaded via a namespace (and not attached):
 [1] igraph_1.0.1       Rcpp_0.12.10       rstudioapi_0.6     magrittr_1.5       munsell_0.4.3      colorspace_1.3-2  
 [7] R6_2.2.0           brew_1.0-6         stringr_1.2.0      plyr_1.8.4         dplyr_0.5.0        visNetwork_1.0.3  
[13] Rook_1.1-1         tools_3.3.1        grid_3.3.1         gtable_0.2.0       DBI_0.6            influenceR_0.1.0  
[19] DiagrammeR_0.9.0   htmltools_0.3.5    lazyeval_0.2.0     digest_0.6.12      assertthat_0.1     tibble_1.2        
[25] gridExtra_2.2.1    RColorBrewer_1.1-2 ggplot2_2.2.1      codetools_0.2-15   htmlwidgets_0.8    viridis_0.3.4     
[31] rgexf_0.15.3       stringi_1.1.3      scales_0.4.1       XML_3.98-1.5       jsonlite_1.3      

## Error Message:
Start training with 1 devices
[01:23:17] /root/mxnet/dmlc-core/include/dmlc/./logging.h:300: [01:23:17] src/ndarray/ndarray.cc:239: Check failed: from.shape() == to->shape() operands shape mismatchfrom.shape = (20,1) to.shape=(20,)

Stack trace returned 10 entries:
[bt] (0) /usr/lib64/R/library/mxnet/libs/libmxnet.so(_ZN4dmlc15LogMessageFatalD1Ev+0x29) [0x7f598889a179]
[bt] (1) /usr/lib64/R/library/mxnet/libs/libmxnet.so(_ZN5mxnet10CopyFromToERKNS_7NDArrayEPS0_i+0x418) [0x7f5988ffb4e8]
[bt] (2) /usr/lib64/R/library/mxnet/libs/libmxnet.so(+0x9b373a) [0x7f5988f9473a]
[bt] (3) /usr/lib64/R/library/mxnet/libs/libmxnet.so(MXImperativeInvoke+0xda2) [0x7f59892c3a62]
[bt] (4) /usr/lib64/R/library/mxnet/libs/mxnet.so(_ZN5mxnet1R7NDArray10CopyFromToERKS1_PS1_+0xa6) [0x7f597b62bed6]
[bt] (5) /usr/lib64/R/library/mxnet/libs/mxnet.so(_ZN5mxnet1R8Executor11UpdateArrayEPKcRKN4Rcpp6VectorILi19ENS4_15PreserveStorageEEEPS7_bb+0x40a) [0x7f597b5efa1a]
[bt] (6) /usr/lib64/R/library/mxnet/libs/mxnet.so(_ZN4Rcpp10CppMethod3IN5mxnet1R8ExecutorEvRKNS_6VectorILi19ENS_15PreserveStorageEEEbbEclEPS3_PP7SEXPREC+0x77) [0x7f597b5f8de7]
[bt] (7) /usr/lib64/R/library/mxnet/libs/mxnet.so(_ZN4Rcpp6class_IN5mxnet1R8ExecutorEE11invoke_voidEP7SEXPRECS6_PS6_i+0xa6) [0x7f597b5fb316]
[bt] (8) /usr/lib64/R/library/Rcpp/libs/Rcpp.so(_Z22CppMethod__invoke_voidP7SEXPREC+0x1c1) [0x7f598c1f9221]
[bt] (9) /usr/lib64/R/lib/libR.so(+0xd40c1) [0x7f59b7c720c1]

Error in exec$update.arg.arrays(arg.arrays, match.name, skip.null) : 
  [01:23:17] src/ndarray/ndarray.cc:239: Check failed: from.shape() == to->shape() operands shape mismatchfrom.shape = (20,1) to.shape=(20,)

Stack trace returned 10 entries:
[bt] (0) /usr/lib64/R/library/mxnet/libs/libmxnet.so(_ZN4dmlc15LogMessageFatalD1Ev+0x29) [0x7f598889a179]
[bt] (1) /usr/lib64/R/library/mxnet/libs/libmxnet.so(_ZN5mxnet10CopyFromToERKNS_7NDArrayEPS0_i+0x418) [0x7f5988ffb4e8]
[bt] (2) /usr/lib64/R/library/mxnet/libs/libmxnet.so(+0x9b373a) [0x7f5988f9473a]
[bt] (3) /usr/lib64/R/library/mxnet/libs/libmxnet.so(MXImperativeInvoke+0xda2) [0x7f59892c3a62]
[bt] (4) /usr/lib64/R/library/mxnet/libs/mxnet.so(_ZN5mxnet1R7NDArray10CopyFromToERKS1_PS1_+0xa6) [0x7f597b62bed6]
[bt] (5) /usr/lib64/R/library/mxnet/libs/mxnet.so(_ZN5mxnet1R8Executor11UpdateArrayEPKcRKN4Rcpp6VectorILi19ENS4_15PreserveStorageEEEPS7_bb+0x40a) [0x7f597b5efa1a]
[bt] (6) /usr/lib64/R/library/mxnet/libs/mxnet.so(_ZN4Rcpp10CppMethod3IN5mxnet1R8ExecutorEvRKNS_6VectorILi19ENS_15PreserveStorageEEEbbEcl

## Minimum reproducible example
model <- mx.model.FeedForward.create(NN_model, 
                                     X                  = trainInput, 
                                     y                  = trainOutput,
                                     ctx                = mx.cpu(),
                                     num.round          = 30,
                                     array.batch.size   = 20,
                                     learning.rate      = 0.05,
                                     momentum           = 0.9,
                                     wd                 = 0.00001,
                                     eval.metric        = mx.metric.accuracy,
                                     epoch.end.callback = mx.callback.log.train.metric(100))

[01:23:17] /root/mxnet/dmlc-core/include/dmlc/./logging.h:300: [01:23:17] src/ndarray/ndarray.cc:239: Check failed: from.shape() == to->shape() operands shape mismatchfrom.shape = (20,1) to.shape=(20,)

Please notice that the "mismatchfrom.shape = (20,1) to.shape=(20,)" corresponds to the "array.batch.size   = 20"

## Steps to reproduce
#!/usr/bin/env Rscript
rm(list = ls())
setwd("/home/master_datasets/")
require(mxnet)
width        = 28
height       = 28
num_channels = 1
train         <- read.csv("train.csv")
test          <- read.csv("test.csv")
train <- data.matrix(train)
test  <- data.matrix(test)
trainInput <- t(train[,-c(1,
                          ncol(train))])
testInput  <- test[,-c(1,
                       ncol(test))]
trainOutput <- t(train[,ncol(train)])
testOutput  <- test[,ncol(test)]
dim(trainInput) <- c(width,
                     height, 
                     num_channels, 
                     ncol(trainInput))
data <- mx.symbol.Variable('data')
conv_1 <- mx.symbol.Convolution(data       = data, 
                                kernel     = c(5,5), 
                                num_filter = 20)
tanh_1 <- mx.symbol.Activation(data     = conv_1, 
                               act_type = "tanh")
pool_1 <- mx.symbol.Pooling(data      = tanh_1, 
                            pool_type = "max", 
                            kernel    = c(2,2), 
                            stride    = c(2,2))
conv_2 <- mx.symbol.Convolution(data       = pool_1,
                                kernel     = c(5,5), 
                                num_filter = 50)
tanh_2 <- mx.symbol.Activation(data     = conv_2, 
                               act_type = "tanh")
pool_2 <- mx.symbol.Pooling(data      = tanh_2, 
                            pool_type = "max", 
                            kernel    = c(2,2), 
                            stride    = c(2,2))
flatData <- mx.symbol.Flatten(data = pool_2)
fcl_1 <- mx.symbol.FullyConnected(data       = flatData,
                                  num_hidden = 500)
tanh_3 <- mx.symbol.Activation(data     = fcl_1, 
                               act_type = "tanh")
fcl_2 <- mx.symbol.FullyConnected(data       = tanh_3, 
                                 num_hidden = 2)
NN_model <- mx.symbol.SoftmaxOutput(data = fcl_2)
mx.set.seed(100)
model <- mx.model.FeedForward.create(NN_model, 
                                     X                  = trainInput, 
                                     y                  = trainOutput,
                                     ctx                = mx.cpu(),
                                     num.round          = 30,
                                     array.batch.size   = 20,
                                     learning.rate      = 0.05,
                                     momentum           = 0.9,
                                     wd                 = 0.00001,
                                     eval.metric        = mx.metric.accuracy,
                                     epoch.end.callback = mx.callback.log.train.metric(100))

## What have you tried to solve it?
No, it seems that is an internal C file error.

Thanks.

**Comment by chavarinbonbyn** [comment]
Please ! any news ?

**Comment by yajiedesign** [comment]
This issue is closed due to lack of activity in the last 90 days. Feel free to reopen if this is still an active issue. Thanks!
